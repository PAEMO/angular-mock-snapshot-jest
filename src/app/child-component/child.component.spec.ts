import {async, ComponentFixture, TestBed} from '@angular/core/testing';

import {ChildComponent} from './child.component';
import {DependencyChildService} from '../dependency-child.service';

const dependencyChildServiceMock = {
  doSomething: jest.fn().mockReturnValue('Return from mock')
};

describe('ChildComponentComponent', () => {
  let component: ChildComponent;
  let fixture: ComponentFixture<ChildComponent>;

  beforeEach(async(() => {
    TestBed.configureTestingModule({
      declarations: [ChildComponent],
      providers: [{provide: DependencyChildService, useValue: dependencyChildServiceMock}]
    })
      .compileComponents();

    fixture = TestBed.createComponent(ChildComponent);
    component = fixture.componentInstance;
    fixture.detectChanges();
  }));

  it('should create', () => {
    expect(component).toBeTruthy();
    expect(dependencyChildServiceMock.doSomething).toHaveBeenCalled();
  });

  it('should render without any input', () => {
    expect(fixture).toMatchSnapshot();
  });

  it('should render with inputs filled and specific mock', () => {
    dependencyChildServiceMock.doSomething.mockReturnValue('Another specific return');

    component.name = 'John';
    component.age = 42;
    fixture.detectChanges(); // detectChanges() is needed here because Inputs are added

    expect(fixture).toMatchSnapshot();
  });
});
