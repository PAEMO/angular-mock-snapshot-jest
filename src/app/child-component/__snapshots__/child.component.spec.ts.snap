// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChildComponentComponent should render with inputs filled and specific mock 1`] = `
<app-child-component
  age={[Function Number]}
  name={[Function String]}
  serviceDependencyChild={[Function Object]}
>
  <p>
    child-component works!
  </p><span>
    From inputs : John is 42
  </span><span>
    Return from service : Another specific return
  </span>
</app-child-component>
`;

exports[`ChildComponentComponent should render without any input 1`] = `
<app-child-component
  serviceDependencyChild={[Function Object]}
>
  <p>
    child-component works!
  </p><span>
    From inputs :  is 
  </span><span>
    Return from service : Return from mock
  </span>
</app-child-component>
`;
